src/pfe/portal/routes/imagePushRegistry.route.js

```diff
@@ -46,6 +46,18 @@ router.post('/api/v1/imagepushregistry', validateReq, async function (req, res)
     const namespace = req.sanitizeBody('namespace');
     const operation = req.sanitizeBody('operation');
 
+    // if user enters it as address/namespace in the UI, throw an error
+    if (namespace.startsWith(address.replace(/\/\s*$/, "") + "/")) {
```

Comment:
I think this check is in the right place now but it's really checking for the one very specific error case where the user has entered the address in the namespace field.
Would it be simpler just to check for characters that are valid for a namespace? That would eliminate the `namespace` as `address/namespace` case *and* a whole load of other invalid cases. Otherwise we're adding an error check that's very specific to one particular problem.
(I'm not sure of the rules for valid `namespace` characters though I've a suspicion it's just the characters valid between `/`'s in a url.)

Reply:
chatted with @hhellyer, will clean up feedback with issue https://github.com/eclipse/codewind/issues/1507 since this requires clean up anyways